yo1k/tic_tac_toe/core/game.py

```diff
@@ -0,0 +1,282 @@
+from __future__ import annotations
+from enum import Enum, auto
+from collections.abc import Sequence
+from typing import Optional
+from abc import ABC, abstractmethod
+
+
+class Mark(Enum):
+    X = auto()
+    O = auto()
+
+
+class Player:
+    def __init__(self, mark: Mark):
+        self.mark = mark
+        self.wins = 0
+
+    def __str__(self):
+        return (
+                f"{type(self)}("
+                f"mark={self.mark},"
+                f"wins={self.wins})")
```

Comment:
If you write it
```python
    def __str__(self):
        return (f"{type(self)}("
                f"mark={self.mark},"
                f"wins={self.wins})")
```
Then you both fewer lines, and separate the type from the attributes. And, just by chance, `return (` is 8 characters long, though this really does not matter for this suggestion. I think this is the way we should format `__str__`.

Reply:
Done.
Your "simple" conventions are not simpler that ordinary rules from the PEP8, where every time I have to think which an option from the rules to use. Now I should remember, that `return` have to be followed by `(` and return value, because this fit vertical alignment (PEP8) and also fit your convention about double indent.